### PR TITLE
fix(tests): remove redundant integration test wf

### PR DESCRIPTION
**Pull request by droctothorpe**
**Description of your changes:**
Identical sample tests are being run in two separate CI workflows for some reason: [kfp-samples](https://github.com/kubeflow/pipelines/blob/master/.github/workflows/kfp-samples.yml) and [kubeflow-pipelines-integration-v2](https://github.com/kubeflow/pipelines/blob/master/.github/workflows/kubeflow-pipelines-integration-v2.yml). The first [invokes](https://github.com/kubeflow/pipelines/blob/c2a77134265dbbe7ca8492e6e429fd3cc60b8419/backend/src/v2/test/sample-test.sh#L28) `sample_test.py`. The second does as [well](https://github.com/kubeflow/pipelines/blob/c2a77134265dbbe7ca8492e6e429fd3cc60b8419/backend/src/v2/test/Makefile#L14) (after pathing through a [Makefile](https://github.com/kubeflow/pipelines/blob/c2a77134265dbbe7ca8492e6e429fd3cc60b8419/backend/src/v2/test/integration-test.sh#L25)). Presumably, we don’t need both. 

I suggest we tackle eliminating `integration-test.sh` and some of the corresponding out of date docs in a follow-up PR. There should be one, clearly documented path for running large tests both in CI and from local.

**Checklist:**
- [x] You have [signed off your commits](https://www.kubeflow.org/docs/about/contributing/#sign-off-your-commits)
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by DharmitD — COMMENTED** [review]
/lgtm

File: .github/workflows/kubeflow-pipelines-integration-v2.yml [hunk 1]
```diff
@@ -1,39 +0,0 @@
...
-  push:
-    branches:
-      - master
-  pull_request:
-    paths:
-      - '.github/workflows/kubeflow-pipelines-integration-v2.yml'
-      - '.github/resources/**'
-      - 'samples/core/parameterized_tfx_oss/**'
-      - 'samples/core/dataflow/**'
-      - 'backend/**'
```
Comment (hbelmiro) [inline .github/workflows/kubeflow-pipelines-integration-v2.yml:13]: We need to add these paths to https://github.com/kubeflow/pipelines/blob/095a794cd46df470a821d60b1b5289f2564d33f3/.github/workflows/kfp-samples.yml#L8 so `kfp-samples.yml` is triggered by changes on those.

**Review by hbelmiro — CHANGES_REQUESTED** [review]
> I suggest we tackle eliminating integration-test.sh

@droctothorpe can you remove the file in this PR.

> and some of the corresponding out of date docs

FYI, We have an open issue for that: https://github.com/kubeflow/pipelines/issues/11061

File: .github/workflows/kubeflow-pipelines-integration-v2.yml (hunk 1 above)
Comment (droctothorpe) [reply .github/workflows/kubeflow-pipelines-integration-v2.yml:13]: Good call. Done. Thanks, @hbelmiro! 🙏

**Review by hbelmiro — COMMENTED** [review]
Thank you for addressing my suggestions @droctothorpe.

> > I suggest we tackle eliminating integration-test.sh
> 
> @droctothorpe can you remove the file in this PR.

You probably missed this comment. Can we remove the file in this PR if it doesn't impact anything?

**Comment by droctothorpe** [comment]
> Thank you for addressing my suggestions @droctothorpe.
> 
> > > I suggest we tackle eliminating integration-test.sh
> > 
> > 
> > @droctothorpe can you remove the file in this PR.
> 
> You probably missed this comment. Can we remove the file in this PR if it doesn't impact anything?

Done! 👍

**Comment by rimolive** [comment]
Based on @hbelmiro suggestions that was already addressed, I believe we're good to go.

/lgtm

**Review by hbelmiro — COMMENTED** [review]
Thank you @droctothorpe.
/lgtm

**Comment by chensun** [comment]
Just to share a bit history, in the past, sample tests and integration tests are different workflows, IIRC, the former runs against the latest release--we do this by maintaining a long running deployment and update it post every release--while the latter runs against the built from master head + the PR being tested--developing on the fly. 
I sense this current GA workflow may have merged, but in the long run, I think there're value to test with both stable (release) and branch head.

**Pull request merged**